File: Revision_classes/06.Dv, LVP,PassByValue/Test4.java
class Test4{
	int x = 10;
	
	void m1() {
		int  x = 50;
		System.out.println(x);
	//	System.out.println(Test4.x);
		System.out.println(this.x);
		
		 x = 60;
		 this.x = 70;
		 System.out.println(x);
		 System.out.println(this.x);
		
	}
	public static void main(String[] args) {
		
		Test4 t1 = new Test4();
		t1.m1();
    //	System.out.println(this.x);
	//	System.out.println(t1.x);
	}
}

File: React/app7/src/Test4.js
export default function Test4() {
    const test1 = (arg1) => {
        console.log('from test1:' + arg1);
    }
    const test2 = (f1, arg1) => {
        console.log('test2 begin: ' +arg1); 
        var results = 100;
        f1(results);
        console.log('test2 end');
    }
    const test3 = (f1, arg1) => {
        console.log('test3 begin:' + arg1); 
        var results = 200;
        f1(results);
        console.log('test3 end');
    }
    const test4 = (f1, arg) => {
        console.log('test4 begin: ' +arg); 
        var results = 300;
        f1(results);
        console.log('test4 end');
    }
    const test5 = () => {
        test2(test1, 100);
        console.log('--------------- ')
        test3((a) => test2(test1, a + 100), 200);
        console.log('--------------- ');
        test4((a) => { test3((b) => test2(test1, a + b + 100), a + 200) }, 300);
    }
    return (
        <>
            <button type="button" onClick={test5}>test</button>
        </>
    );
}